File: wiki_examples/src/main/java/com/github/klee0kai/wiki/provide/binding/GodWorkspaceComponent.java
package com.github.klee0kai.wiki.provide.binding;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.test.mowgli.galaxy.Sun;

@Component
public interface GodWorkspaceComponent {

    SunSystemModule sunSystem();

    Sun sun();

}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/provide/binding/GodWorkspaceComponent.kt
package com.github.klee0kai.wiki.provide.binding

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.test.mowgli.galaxy.Sun

@Component
interface GodWorkspaceComponent {

    fun sunSystem(): SunSystemModule

    fun sun(): Sun

}